scripts/debug/novatel_client.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+"""
+Novatel Client module.
+
+This script connects to a Novatel GNSS receiver, configures it, and logs the
+received data.
+"""
+
+import argparse
+import logging
+import socket
+
+import colorlog
+
+
+class NovatelClient:
+    """Client to interact with a Novatel GNSS receiver."""
+
+    def __init__(self, gnss_ip, gnss_port):
+        """Initialise NovatelClient with the GNSS receiver's IP and port."""
+        self.gnss_ip = gnss_ip
+        self.gnss_port = gnss_port
+        self.gnss_socket = None
+        self.port_name = None
+
+    def connect(self):
+        """Connect to the GNSS receiver via TCP."""
+        try:
+            self.gnss_socket = socket.socket(
+                socket.AF_INET, socket.SOCK_STREAM
+            )
+            self.gnss_socket.connect((self.gnss_ip, self.gnss_port))
+            logging.info(
+                f'Connected to GNSS receiver at '
+                f'{self.gnss_ip}:{self.gnss_port}'
+            )
+
+        except (OSError, socket.timeout) as e:
+            logging.error(f'Failed to connect: {e}')
+
+    def configure_port(self):
+        """Configure the GNSS port to log NMEA data."""
+        configure_command = (
+            '\r\n'
+            'unlogall thisport\r\n'
+            'log gpggalong ontime 0.1\r\n'
+            'log gprmc ontime 0.1\r\n'
+            'log gpgst ontime 0.2\r\n'
+            'interfacemode rtcmv3 novatel\r\n'  # Set RX and TX
+        )
+
+        try:
+            self.gnss_socket.sendall(configure_command.encode('utf-8'))
+            logging.info(
+                f'Sent configuration commands to port: {self.gnss_port}'
+            )
+
+            # Read the response to confirm configuration
+            response = self.gnss_socket.recv(1024).decode('utf-8')
```

Comment:
Magic number? This should be a constant somewhere with a clear name to what this is for.

Reply:
`ntrip_client.py` removed